docs/advanced/repo_subdirectories.rst

```diff
@@ -0,0 +1,17 @@
+Organizing cookiecutter templates in subdirectories
+===================================================
+
+You can also organize one or more templates in a single repo, like this:
+
+    https://github.com/user/repo-name.git
+        ├── subdirectory-name/
+            ├── {{cookiecutter.project_slug}}/
+            └── cookiecutter.json
+
+And then, using the cli
+
+    cookiecutter https://github.com/user/repo-name.git --subdirectory="subdirectory-name"
+
+
+Cookiecutter clones the repo into `~/.cookiecutter` as usual, then `cd`s
```

Comment:
This does not explain what user get's at the end. As far as I can understand it is half way only.

Reply:
Will it work wiz zip files?

Reply:
I'll reword it to be more explicit.  Yup, it works with local, zip, github (updated OP with more checkboxes)